libcontainer/init_linux.go

```diff
@@ -273,6 +273,36 @@ func syncParentHooks(pipe io.ReadWriter) error {
 	return readSync(pipe, procResume)
 }
 
+// syncParentSeccomp sends to the given pipe a JSON payload which
+// indicates that the parent should pick up the seccomp fd with pidfd_getfd()
+// and send it to the seccomp agent over a unix socket. It then waits for
+// the parent to indicate that it is cleared to resume and closes the seccompFd.
+// If the seccompFd is -1, there isn't anything to sync with the parent, so it
+// returns no error.
+func syncParentSeccomp(pipe io.ReadWriter, seccompFd int) error {
+	if seccompFd == -1 {
+		return nil
+	}
+
+	// Tell parent.
+	if err := writeSyncWithFd(pipe, procSeccomp, seccompFd); err != nil {
+		unix.Close(seccompFd)
+		return err
+	}
+
+	// Wait for parent to give the all-clear.
+	if err := readSync(pipe, procSeccompDone); err != nil {
+		unix.Close(seccompFd)
+		return fmt.Errorf("sync parent seccomp: %w", err)
+	}
+
+	if err := unix.Close(seccompFd); err != nil {
```

Comment:
Do we need to retry on EINTR? (for every `unix.Close`)

Reply:
I don't think so. Because:
 * If it fails with EINTR, we will exit, so it is not that it is not handled. The container might fail in the worst case, but not run with the seccomp fd open, that would be a big problem. If there are bug reports about this, of course, it can be changed to loop instead
 * IIUC, it should not return EINTR, as go runtime registers the signal handlers with SA_RESTART. That means that the close(2) syscall should be automatically restarted if it was interrupted, instead of returning EINTR. There are bugs in the wild, though, and some syscalls on some OS do return EINTR when they shouldn't. This should be only Linux and I haven't seen changes in the go runtime looping on close to handle those bugs (as they were done with others), so I think we are just fine.

But, as you prefer, really.